rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/Sw360LicenseService.java

```diff
@@ -161,4 +170,26 @@ public void importSpdxInformation(User sw360User) throws TException {
             throw new HttpMessageNotReadableException("Unable to import All Spdx license. User is not admin");
         }
     }
+
+    public void uploadLicense(User sw360User, MultipartFile file, HttpServletRequest request) throws IOException , TException {
+		final HashMap<String, InputStream> inputMap = new HashMap<>();
+
+		if (!PermissionUtils.isUserAtLeast(UserGroup.ADMIN, sw360User)) {
+			throw new HttpMessageNotReadableException("Unable to upload license file. User is not admin");
+		}
+        try {
+            InputStream inputStream = file.getInputStream();
+            ZipTools.extractZipToInputStreamMap(inputStream, inputMap);
+            LicenseService.Iface sw360LicenseClient = getThriftLicenseClient();
+            boolean overwriteIfExternalIdMatches = "true".equals(request.getParameter("overwriteIfExternalIdMatches"));
```

Comment:
Use @RequestParam instead of request.getParameter

Reply:
done.